File: Cryptofolio/Models/Watchlist.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Cryptofolio.Models
{
    public class Watchlist
    {
        public Watchlist()
        {
            this.Coins = new List<Coin>();
        }

        public int Id { get; set; }

        public string? ApplicationUserId { get; set; }

        [ForeignKey("ApplicationUserId")]
        public ApplicationUser ApplicationUser { get; set; }

        public List<Coin> Coins { get; set; }

    }
}

File: Cryptofolio/ClientApp/src/app/common/models/watchlist.ts
import { ApplicationUser } from "./application-user";
import { Coin } from "./coin";

export class Watchlist {

  public Id : number = -1;
  public ApplicationUserId : string = '';
  // public ApplicationUser!: ApplicationUser;
  public Coins: Coin[] = [];

  constructor(Id: number, Coins: Coin[], ApplicationUserId : string) {
    this.Id = Id;
    this.Coins = Coins;
    this.ApplicationUserId = ApplicationUserId;
  }
}


export class AddCoinToWatchlistRequest {
  public WatchlistId : number = -1;
  public CoinSymbol : string = "";
  public StarIndicator : boolean = false;
}
